### PR TITLE
Migrating to Elasticsearch

### PR DESCRIPTION
**Pull request by Rub21**
I've updated the backend, almost all endpoint which I did use Postgres are working using elasticSearch.

to check it out: use ElasticSearch locally or ask to me for an ElasticSearch endpoint in the cloud, which contain already some tasks

cc. @rodowi @aaronlidman

### CODE REVIEW

File: src/utils/loadGeojsonToDB.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict';
```
Comment (aaronlidman) [inline src/utils/loadGeojsonToDB.js]: Is this file still needed?

File: src/utils/saveGeojsonFromDB.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+var fs = require('fs');
+var pg = require('pg');
+var copyTo = require('pg-copy-streams').to;
+var config = require('./../configs/config');
+
+module.exports.saveBackup = function(task, cb) {
```
Comment (defvol) [inline src/utils/saveGeojsonFromDB.js]: @aaronlidman @Rub21 can we use this script to migrate tasks from the old stack?

File: src/controllers/ControllerItems.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  });
+};
+
+module.exports.getAllItems = function(request, reply) {
+  var idtask = request.params.idtask;
+  client.search({
+    index: 'tofix',
+    type: idtask,
+    body: {
+      size: 1000, //get item using pagination - need to fix here
```
Comment (defvol) [inline src/controllers/ControllerItems.js]: We could probably default this to a lower number, like 50 or 100, to keep the frontend responsive. RE pagination: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-from-size.html

File: src/controllers/ControllerTasks.js [hunk 4]
```diff
@@ -126,22 +126,38 @@ module.exports.createTasks = function(request, reply) {
...
+              maxRetries: 5,
+              index: 'tofix',
+              id: task.idtask,
+              type: task.idtask,
+              body: bulkChunk
+            }, function(err) {
+              if (err) {
+                //try 3 times to save the chunk
+                errorsCounter++;
+                if (errorsCounter === 3) {
```
Comment (bkowshik) [inline src/controllers/ControllerTasks.js:171]: @Rub21 Why are we expecting it to succeed on a retry after failure? Is this how ElasticSearch works?

Comment (Rub21) [reply src/controllers/ControllerTasks.js:171]: @bkowshik We are sending chunks of data, there is little possibility of failure when it sends the data, to avoid this possibility of failure we are trying to send it three times.
- The failure could be by internet connection or  when the features of geojson are quite big.
